src/convenience/session.ts

```diff
@@ -314,10 +310,24 @@ export function lazySession<S, C extends Context>(
     };
 }
 
+/** Stores session data per chat by default */
 function defaultGetSessionKey(ctx: Context): string | undefined {
     return ctx.chat?.id.toString();
 }
 
+/** Returns a useful error message for when the session key is undefined */
+function undef<C extends Context>(
+    op: "access" | "assign",
+    getSessionKey: SessionOptions<C>["getSessionKey"],
+    opts: { lazy?: boolean } = {},
+) {
+    const lazy = opts.lazy ?? false;
+    const reason = getSessionKey === defaultGetSessionKey
+        ? "this update does not belong to a chat, so the session key is undefined"
+        : "your custom `getSessionKey` function returned undefined for this update";
```

Comment:
in some cases I send the `error.message` to the user and print the full error on the console. This way the user can have some idea what went wrong in some cases. Is it my fault or is it something wrong in the bot. I even got a „this is your problem, fix Line n“ as a PM once because of this. Using „your“ in this case can be confusing.

```suggestion
        : "the custom `getSessionKey` function returned undefined for this update";
```

Reply:
I would never have though about that. Thanks